front/src/applications/operationalStudies/components/SimulationResults/ChartHelpers/defineChart.ts

```diff
@@ -63,7 +63,7 @@ const defineChart = (
     .append('g')
     .attr('transform', `translate(0,${height})`)
     .attr('class', 'grid')
-    .call(gridX(x, height));
+    .call(gridX(x as d3.ScaleTime<number, number>, height));
```

Comment:
No need to cast here. Moreover, sometimes x is not a ScaleTime but a LinearScale
```suggestion
    .call(gridX(x, height));
```

Reply:
done